includes/validation/class-amp-validation-manager.php

```diff
@@ -1864,7 +1887,7 @@ public static function finalize_validation( Document $dom ) {
 
 		// Otherwise, since it is in a paired mode, only allow showing the dirty AMP page if the user is authorized.
 		// If not, normally the result is redirection to the non-AMP version.
-		return self::has_cap() || current_user_can( 'manage_options' ) || is_customize_preview();
+		return self::has_cap() || is_customize_preview(); // @todo do self::get_dev_tools_user_access()->is_user_enabled() instead of has_cap?
```

Comment:
I think `self::has_cap()` is sufficient here. I don't see any reason as to why the dirty AMP page should not be shown if the user does not have developer tools enabled.

Reply:
Meaning without `is_customize_preview()`? That's needed to ensure that someone can access an AMP page in the Customizer even if it has kept invalid markup.

The `self::get_dev_tools_user_access()->is_user_enabled()` method here includes `self::has_cap()`, but also takes into account whether the user has dev tools enabled.

Reply:
> Meaning without is_customize_preview()?

Yes I meant `self::has_cap()` along with `is_customize_preview()`.

---

> The self::get_dev_tools_user_access()->is_user_enabled() method here includes self::has_cap(), but also takes into account whether the user has dev tools enabled.

Right. I guess the question can be reduced to: Should we **not** show the dirty AMP page if the user has developer tools enabled? 

In that case I'd vote no, as if that were the case, a user that has the `amp_validate` capability but not dev tools enabled would not be able to see the dirty AMP page.

Reply:
Actually, yes, using `has_cap` should be used instead of `is_user_enabled` because a user who has the capability can still access dev tools.

```suggestion
		return self::has_cap() || is_customize_preview();
```